src/main/java/org/jenkinsci/maven/plugins/hp/util/Utils.java

```diff
@@ -0,0 +1,22 @@
+package org.jenkinsci.maven.plugins.hp.util;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class Utils {
+
+    /**
+     * Return an unmodifiable set whose contents are the unions of all the specified sets
+     * @param <T>
+     */
+    @SafeVarargs
+    public static <T> Set<T> unionOf(Set<T>... sets) {
+        HashSet<T> unionSet = new HashSet<>(sets[0]);
```

Comment:
Effective Java Item 52: Refer to objects by their interfaces

```suggestion
        Set<T> unionSet = new HashSet<>(sets[0]);
```